### PR TITLE
boxes: init at 1.3

### PR DESCRIPTION
**Pull request by waiting-for-dev**
###### Motivation for this change

Adds [boxes](https://boxes.thomasjensen.com/) package.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

###### Comments

This is my first derivation, hope everything is fine. I have some doubts about the issue with the system wide config file used in boxes. The file is defined in the `Makefile` and defaults to `/usr/share/boxes`. In the derivation I'm changing it for `$out/boxes-config`. Individual users can define its own config file but, anyway, I'm not sure if there is a better way to go. Also, it would be nice to set the new config file path in a `mkDerivation` attribute, so that it can be overriden easily if needed. However, if I'm not wrong, I have no way to access `$out` from there, so it's not possible.

### CODE REVIEW

File: pkgs/tools/text/boxes/default.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  src = fetchFromGitHub {
+    owner = "ascii-boxes";
+    repo = "boxes";
+    rev = "v${version}";
+    sha256 = "0b12rsynrmkldlwcb62drk33kk0aqwbj10mq5y5x3hjf626gjwsi";
+  };
+
+  # Building instructions:
+  # https://boxes.thomasjensen.com/build.html#building-on-linux--unix
+  buildInputs = [ bison flex ];
```
Comment (marsam) [inline pkgs/tools/text/boxes/default.nix]:
```suggestion
  nativeBuildInputs = [ bison flex ];
```

File: pkgs/tools/text/boxes/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  dontConfigure = true;
+
+  # Makefile references a system wide config file in '/usr/share'. Instead, we
+  # move it within the store by default.
+  preBuild = ''
+    sed -i "/^GLOBALCONF = \/usr\/share\/boxes$/c\GLOBALCONF=$out/share/boxes-config" Makefile
```
Comment (marsam) [inline pkgs/tools/text/boxes/default.nix]: If we use `substituteInPlace` we don't have to worry about escaping
```suggestion
    substituteInPlace Makefile \
      --replace "GLOBALCONF = /usr/share/boxes" "GLOBALCONF=${placeholder "out"}/share/boxes/boxes-config"
```

File: pkgs/tools/text/boxes/default.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  '';
+
```
Comment (marsam) [inline pkgs/tools/text/boxes/default.nix]: darwin needs:
```suggestion
makeFlags = stdenv.lib.optionals stdenv.isDarwin [ "CC=cc" ];
```

File: pkgs/tools/text/boxes/default.nix [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  installPhase = ''
+    mkdir -p $out/{bin,share}
```
Comment (marsam) [inline pkgs/tools/text/boxes/default.nix]: let's use `install`:
```suggestion
    install -Dm755 -t $out/bin src/boxes
    install -Dm644 -t $out/share/boxes boxes-config
    install -Dm644 -t $out/share/man/man1 doc/boxes.1
```
with this, we can remove the `postFixup` below

File: pkgs/tools/text/boxes/default.nix [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+  postFixup = ''
+    cp boxes-config doc/boxes.1 $out/share
+    cp src/boxes $out/bin
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Command line ASCII boxes unlimited!";
+    longDescription = ''
+        Boxes is a command line filter program that draws ASCII art boxes around your input text.
```
Comment (marsam) [inline pkgs/tools/text/boxes/default.nix]: very minorly
```suggestion
      Boxes is a command line filter program that draws ASCII art boxes around your input text.
```

File: maintainers/maintainer-list.nix [hunk 6]
```diff
@@ -7652,4 +7652,10 @@
     githubId = 50867187;
     name = "Rakesh Gupta";
   };
+  waiting-for-dev = {
```
Comment (marsam) [inline maintainers/maintainer-list.nix]: please put this in a separate commit

**Review by marsam — COMMENTED** [review]
Thanks for your contribution! I left some comments, but LGTM overall

**Comment by waiting-for-dev** [comment]
Thanks a lot for your review @marsam. I addressed all your comments and forced the push in two commits as you recommended.

**Comment by marsam** [comment]
Thanks!

**Comment by waiting-for-dev** [comment]
Thanks for your help @marsam :+1:
